js/libs/event_dispatcher.js

```diff
@@ -0,0 +1,217 @@
+/*global Map, Set */
```

Comment:
simple lib we use in SMS app (that's in /shared now :) ) to implement a simple event mechanism.

Gabriele wrote a David Flanagan-style documentation :)